server/src/test/java/org/apache/druid/server/coordinator/DataSourceCompactionConfigTest.java

```diff
@@ -83,6 +84,7 @@ public void testSerdeBasic() throws IOException
     Assert.assertEquals(config.getTuningConfig(), fromJson.getTuningConfig());
     Assert.assertEquals(config.getTaskContext(), fromJson.getTaskContext());
     Assert.assertEquals(config.getGranularitySpec(), fromJson.getGranularitySpec());
+    Assert.assertEquals(config.getEngine(), fromJson.getEngine());
```

Comment:
There should be another assertion that the engine is actually `native`, since `null` had been passed to the constructor.

Reply:
The engine will be null as the override with default happens at the submission time only.